packages/logger/src/logger.ts

```diff
@@ -0,0 +1,97 @@
+import { LogEvent } from './event';
+import { LOG_LEVEL, LogLevel } from './levels';
+import { Transport } from './transport';
+import { Transformer } from './transformer';
+
+export type LoggerConfig = {
+  logLevel: LogLevel;
+  autoFlush: boolean;
+  transport: Transport;
+  transformers: Transformer[];
+  args?: { [key: string]: any };
+};
+
+export class Logger {
+  children: Logger[] = [];
+
+  constructor(public config: LoggerConfig) {
+    if (this.config.logLevel == undefined || this.config.logLevel < 0) {
+      this.config.logLevel = LogLevel[LOG_LEVEL as keyof typeof LogLevel];
+    }
+  }
+
+  debug = (message: string, args: { [key: string]: any } = {}) => {
+    this._log(LogLevel.debug, message, args);
+  };
+  info = (message: string, args: { [key: string]: any } = {}) => {
+    this._log(LogLevel.info, message, args);
+  };
+  warn = (message: string, args: { [key: string]: any } = {}) => {
+    this._log(LogLevel.warn, message, args);
+  };
+  error = (message: string, args: { [key: string]: any } = {}) => {
+    this._log(LogLevel.error, message, args);
+  };
+  fatal = (message: string, args: { [key: string]: any } = {}) => {
+    this._log(LogLevel.fatal, message, args);
+  };
+
+  with = (args: { [key: string]: any }) => {
+    const config = { ...this.config, args: { ...this.config.args, ...args } };
+    const child = new Logger(config);
+    this.children.push(child);
+    return child;
+  };
+
+  _log = (level: LogLevel, message: string, args: { [key: string]: any } = {}) => {
```

Comment:
nit: any reason not to use `private _log` or `#log` for this instead of `_log`?

Reply:
no reason, added it.